test/conformance/ptp/ptp.go

```diff
@@ -356,14 +370,102 @@ var _ = Describe("[ptp]", func() {
 							buf, _ := pods.ExecCommand(client.Client, pod, PtpContainerName, []string{"curl", "127.0.0.1:9091/metrics"})
 							return buf.String()
 						}, 5*time.Minute, 5*time.Second).Should(ContainSubstring("openshift_ptp_offset_ns"),
-							fmt.Sprint("Time metrics are not detected"))
+							"Time metrics are not detected")
 						slavePodDetected = true
 						break
 					}
 				}
 				Expect(slavePodDetected).ToNot(BeFalse(), "No slave pods detected")
 			})
 		})
+
+		Context("Running with event enabled", func() {
+			ptpSlaveRunningPods := []v1core.Pod{}
+			BeforeEach(func() {
+				if !ptpEventEnabled() {
+					Skip("Skipping, PTP events not enabled")
+				}
+				if isSingleNode {
+					Skip("Running in single node mode")
+				}
+				if discoveryFailed {
+					Skip("Failed to find a valid ptp slave configuration")
+				}
+				ptpPods, err := client.Client.Pods(PtpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(ptpPods.Items)).To(BeNumerically(">", 0), "linuxptp-daemon is not deployed on cluster")
+
+				for _, pod := range ptpPods.Items {
+					if podRole(pod, slaveNodeLabel) {
+						waitUntilLogIsDetected(pod, 3*time.Minute, "Profile Name:")
+						ptpSlaveRunningPods = append(ptpSlaveRunningPods, pod)
+					}
+				}
+				if discovery.Enabled() {
+					Expect(len(ptpSlaveRunningPods)).To(BeNumerically(">=", 1), "Fail to detect PTP slave pods on Cluster")
+				} else {
+					Expect(len(ptpSlaveRunningPods)).To(BeNumerically(">=", 1), "Fail to detect PTP slave pods on Cluster")
+				}
+			})
+
+			It("Should check for ptp events ", func() {
+				By("Checking event side car is present")
+
+				for _, pod := range ptpSlaveRunningPods {
+					cloudProxyFound := false
+					Expect(len(pod.Spec.Containers)).To(BeNumerically("==", 3), "linuxptp-daemon is not deployed on cluster with cloud event proxy")
```

Comment:
Any reason this check is necessary?  This will break if adding another container at some point, right?

Reply:
This is valid for event proxy . Right now we have 3 containers  when event is enabled , that is we run event proxy as side car  . When event is not running we just have 2 containers .
Adding another container will be major changes which needs to revisit this and I don’t think we will be adding another sidecar to this .